src/components/request/Request.tsx

```diff
@@ -14,12 +16,25 @@ import RequestHero from "./RequestHero";
 */
 
 const Request = () => {
+  const client = useContext(RequestClientContext);
   const [searchParams] = useSearchParams();
   const { rows, headerCells } = useRequestTableData(searchParams);
 
+  useEffect(() => {
+    const request = {
+      requester: searchParams.get("requester") ?? "",
+      identifier: searchParams.get("identifier") ?? "",
+      timestamp: Number(searchParams.get("timestamp")) ?? "",
+      ancillaryData: searchParams.get("ancillaryData") ?? "",
+      chainId: Number(searchParams.get("chainId")) ?? 1,
+    };
+
+    client.setActiveRequest(request);
```

Comment:
> This question is more for @daywiss and not related to this code, but what happens exactly when this function is called? How does internal state changes?

so when the client knows of a user and a particular request it does a few things:
1. looks up request state from the OO
2. looks up default liveness from oo
3. looks up users collateral balance based on request.currency
4. looks up oracles allowance to user using request.currency

This data gets cached into a singular global state object that you can then use client.store.read which curates an api for reading data from the store

This is everything you need to know to start proposing, or to disable buttons/show errors. Eventually it will also handle the transaction process for both approving and submitting proposals/disputes.